src/sentry/workflow_engine/types.py

```diff
@@ -32,17 +33,19 @@ class DetectorPriorityLevel(IntEnum):
 ProcessedDataConditionResult = tuple[bool, list[DataConditionResult]]
 
 
-class EventJob(TypedDict):
+@dataclass(frozen=True)
+class EventJob:
     event: GroupEvent
 
 
-class WorkflowJob(EventJob, total=False):
-    group_state: GroupState
-    is_reprocessed: bool
-    has_reappeared: bool
-    has_alert: bool
-    has_escalated: bool
-    workflow: Workflow
+@dataclass(frozen=True)
+class WorkflowJob(EventJob):
+    group_state: GroupState | None = None
+    is_reprocessed: bool | None = None
+    has_reappeared: bool | None = None
+    has_alert: bool | None = None
```

Comment:
i don't think we'll need this attribute either - afaik we're not setting this up as a replacement with detectors, instead the occurrence itself knows if it was created by a detector or not.

Reply:
removing